src/config/logger.ts

```diff
@@ -0,0 +1,21 @@
+import winston from 'winston';
+
+const logger = winston.createLogger({
+  level: 'info',
+  format: winston.format.combine(
```

Comment:
I would use structured logging format. It's basically log everything as JSON object. Structured logs can be parsed by 3rd party logs system: AWS CloudWatch, for example. It allows you to filter logs by some nested fields
```suggestion
  format: winston.format.combine(
    winston.format.timestamp(),
    winston.format.json()
  ),
```

Reply:
In this case, I'm not sure that you need abstractions like requestSent and requestError. You can use logger directly

Reply:
example, logging all incoming requests

`logger.info(`Invoking request ${method} ${url}`, {method, url, body});`

output

```json
{
  "body": { "age": 30, "login": "llllll", "password": "********" },
  "level": "info",
  "message": "Invoking request POST /users",
  "method": "POST",
  "timestamp": "2023-03-06T09:43:14.778Z",
  "url": "/users"
}
```

Reply:
as I said, 3rd party logging systems understand json, and, for exmaple, you can apply filtering on json fields (find all logs where body.age = 30)

Reply:
used json format instead of string